### PR TITLE
Update `mini_racer` to 0.8.0

### PR DESCRIPTION
**Pull request by Hamms**
Now that we're on Ubuntu 20 and so can successfully install the `libv8-node` dependency (Ubuntu 18 had issues with incompatible gcc versions), we can not only update `mini_racer` to the latest version but also remove the temporary logic we put in place to avoid trying to install it in development environments for the sake of MacOS users. And also clean up some no-longer-necessary workaround instructions from `SETUP.md`!

## Links

- https://github.com/code-dot-org/code-dot-org/pull/47788
- https://codedotorg.slack.com/archives/C0T0PNTM3/p1680816473255759?thread_ts=1680568332.981739&cid=C0T0PNTM3

## Testing story

Verified on [an adhoc](https://adhoc-mini-racer-0-8-0.cdn-code.org) that Ubuntu 20 can indeed install this version.

Also tested on MacOS; @carl-codeorg [verified that he was able to `bundle install` successfully](https://codedotorg.slack.com/archives/C03CK49G9/p1705704939310459?thread_ts=1705704441.299019&cid=C03CK49G9). Thanks, Carl!

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
Thank you for doing this! It looks like there are also some mentions of mini_racer in SETUP.md that can now be removed as well.

**Comment by Hamms** [comment]
Oh, good catch, thanks! I'll take care of those in this PR, too

File: SETUP.md [hunk 1]
```diff
@@ -543,16 +499,6 @@ If you run into an error like "Don't know how to set rpath on your system, if My
 
 - `gem install mysql2 -v '0.5.2' -- --with-opt-dir=$(brew --prefix openssl) --with-ldflags=-L/opt/homebrew/Cellar/zstd/1.5.0/lib`
 
-#### therubyracer
-
-If you run into an issue about therubyracer while running `bundle install` try :
```
Comment (Hamms) [inline SETUP.md:548]: Note that although this PR is not removing `therubyracer`, I'm unable to find any other references to it in the codebase, and based on the linked SO question it seems like this is actually an issue not with `therubyracer` itself but with `libv8`, which this PR _is_ removing.

File: SETUP.md [hunk 2]
```diff
@@ -404,44 +404,8 @@ Wondering where to start?  See our [contribution guidelines](CONTRIBUTING.md) fo
...
-git update-index --assume-unchanged Gemfile.lock
-git update-index --no-assume-unchanged Gemfile.lock
-git ls-files -v | grep '^[[:lower:]]'
-```
-</details>
-
-
-
-Then run the following commands to successfully complete a bundle install:
+Make you sure you've already run ```bundle config --local without staging test production levelbuilder``` command
```
Comment (davidsbailey) [inline SETUP.md]: it should be possible to fully remove `bundle config --local without` from SETUP.md (here and above), now that there are no longer any gems which we exclude from development.

Comment (Hamms) [reply SETUP.md]: Awesome!

**Review by davidsbailey — APPROVED** [review]
woohoo!

**Pull request merged**